control_unit.py

```diff
@@ -67,22 +67,33 @@ def _STR(instruction):
 
 
 def _RTI(instruction):
-    """unused"""
-    pass  # TODO throw exception here.
+    raise Exception("instruction is not implemented !")
+    pass  
 
 
 def _NOT(instruction):
-    """bitwise not"""
+    DR = (_NOT(instruction) >> 9) & 0X7                            # Destenation Register.
+    SR = (_NOT(instruction) >> 6) & 0X7                            # Source Register (register cntaining the data).
+    for B in range (16):                                           # B stands (in this line) for Bit.
```

Comment:
it seems like you're `inverting` each bit separately, you don't need to do that, Python (like almost every language) allows you to manipulate the variable at once:
`reg_write(Registers(SR), ~reg_read(Registers(DR)))`

Reply:
dig it